File: version_2/version_2/Candidate.cpp
/***********************************************************************
* Module:  Candidate.cpp
* Author:  Anton
* Modified: Monday, November 7, 2016 6:50:33 PM
* Purpose: Implementation of the class Candidate
***********************************************************************/

#include <vector>
#include <string>
#include "Resume.h"
#include "Candidate.h"
#include "Technical_manager.h"

Candidate::Candidate() :
	commutabilityLevel(1),
	worryingLevel(2),
	work(false),
	resume()
{
	std::cout << "Candidate was created (default)" << std::endl;
}

Candidate::Candidate(int cLevel, int wLevel, bool work):
	commutabilityLevel(cLevel),
	worryingLevel(wLevel),
	work(work),
	resume()
{
	//resume = new Resume;
	std::cout << "Candidate was created (inicialisation)" << std::endl;
}

Candidate::Candidate(Candidate &sCandidate):
	commutabilityLevel(sCandidate.commutabilityLevel),
	worryingLevel(sCandidate.worryingLevel),
	work(sCandidate.work),
	resume()
{
	//resume = new Resume;
	std::cout << "Candidate was created (copy)" << std::endl;
}

Candidate::Candidate(std::string str):
	resume(),
	commutabilityLevel(atoi(str.substr(0, 2).c_str())),
	worryingLevel(atoi(str.substr(2, 2).c_str())),
	work(atoi(str.substr(4, 1).c_str()))
{
	//commutabilityLevel = atoi(str.substr(0, 2).c_str());
	//worryingLevel = atoi(str.substr(2, 2).c_str());
	//work = atoi(str.substr(4, 1).c_str());
	std::cout << "Candidate was ceated (transform)" << std::endl;
}

Candidate::~Candidate()
{
	//delete resume;
	std::cout << "Candidate was destroyed" << std::endl;
	std::cout << std::endl;
	//std::cout << std::endl;
}
////////////////////////////////////////////////////////////////////////
// Name:       Candidate::prepareResume()
// Purpose:    Implementation of Candidate::prepareResume()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Candidate::prepareResume(void)
{
	// TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Candidate::takeTest()
// Purpose:    Implementation of Candidate::takeTest()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Candidate::takeTest(void)
{
	// TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Candidate::passInterview()
// Purpose:    Implementation of Candidate::passInterview()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Candidate::passInterview(void)
{
	// TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Candidate::get()
// Purpose:    Implementation of Candidate::get()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Candidate::getWork(void)
{
	// TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Candidate::get()
// Purpose:    Implementation of Candidate::get()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Candidate::get(void)
{
	// TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Candidate::set()
// Purpose:    Implementation of Candidate::set()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Candidate::set(void)
{
	// TODO : implement
}

File: cs_version2/cs_version2/Candidate.cs
/***********************************************************************
 * Module:  Candidate.cs
 * Author:  Anton
 * Purpose: Definition of the Class Candidate
 ***********************************************************************/

using System;

public class Candidate
{
    public Candidate()
    {
        Resume resume = new Resume();
        commutabilityLevel = 1;
        worryingLevel = 2;
        work = false;
        Console.WriteLine("Candidate was created (default)");
    }
	public Candidate(int cLevel, int wLevel, bool work)
    {
        Resume resume = new Resume();
        commutabilityLevel = cLevel;
        worryingLevel = wLevel;
        work = work;
        Console.WriteLine("Candidate was created (inicialisation)");
    }
    public Candidate(Candidate sCandidate)
    {
        resume = sCandidate.resume;
        commutabilityLevel = sCandidate.commutabilityLevel;
        worryingLevel = sCandidate.worryingLevel;
        work = sCandidate.work;
        Console.WriteLine("Candidate was created (copy)");
    }
   public void prepareResume()
   {
      // TODO: implement
   }
   
   public void takeTest()
   {
      // TODO: implement
   }
   
   public void passInterview()
   {
      // TODO: implement
   }
   
   public void get()
   {
      // TODO: implement
   }
   
   public void set()
   {
      // TODO: implement
   }
   
   public void getWork()
   {
      // TODO: implement
   }

   public System.Collections.ArrayList resume;
   
   /// <pdGenerated>default getter</pdGenerated>
   public System.Collections.ArrayList GetResume()
   {
      if (resume == null)
         resume = new System.Collections.ArrayList();
      return resume;
   }
   
   /// <pdGenerated>default setter</pdGenerated>
   public void SetResume(System.Collections.ArrayList newResume)
   {
      RemoveAllResume();
      foreach (Resume oResume in newResume)
         AddResume(oResume);
   }
   
   /// <pdGenerated>default Add</pdGenerated>
   public void AddResume(Resume newResume)
   {
      if (newResume == null)
         return;
      if (this.resume == null)
         this.resume = new System.Collections.ArrayList();
      if (!this.resume.Contains(newResume))
         this.resume.Add(newResume);
   }
   
   /// <pdGenerated>default Remove</pdGenerated>
   public void RemoveResume(Resume oldResume)
   {
      if (oldResume == null)
         return;
      if (this.resume != null)
         if (this.resume.Contains(oldResume))
            this.resume.Remove(oldResume);
   }
   
   /// <pdGenerated>default removeAll</pdGenerated>
   public void RemoveAllResume()
   {
      if (resume != null)
         resume.Clear();
   }

   private int commutabilityLevel;
   private int worryingLevel;
   private bool work;

}